nlmod/gwf/wells.py

```
@@ -159,3 +181,69 @@ def maw_from_df(
     )
 
     return maw
+
+
+def _add_cellid(df, ds=None, gwf=None, x=None, y=None):
+    if not isinstance(df, gpd.GeoDataFrame):
+        df = gpd.GeoDataFrame(df, geometry=gpd.points_from_xy(df[x], df[y]))
+    if "cellid" not in df.columns:
+        df = gdf_to_grid(df, gwf if ds is None else ds)
+    return df
+
+
+def _get_layer_multiplier_for_wells(df, top, botm, ds=None, gwf=None):
+    # get required data either from  gwf or ds
+    if ds is None:
+        ml_top = gwf.dis.top.array
+        ml_bot = gwf.dis.botm.array
+        kh = gwf.npf.k.array
+        layer = range(gwf.dis.nlay.array)
+    else:
```

maybe raise an Exception if both are None?